File: stepik-union/src/main/java/org/stepik/plugin/projectWizard/ui/ProjectSetting.java
package org.stepik.plugin.projectWizard.ui;

import org.stepik.core.SupportedLanguages;
import org.jetbrains.annotations.NotNull;
import org.stepik.api.objects.StudyObject;

/**
 * @author meanmail
 */
interface ProjectSetting {
    void selectedStudyNode(@NotNull StudyObject course);

    void addListener(@NotNull ProjectSettingListener listener);

    void removeListener(@NotNull ProjectSettingListener listener);

    void selectedProgrammingLanguage(@NotNull SupportedLanguages language);
}

File: stepik-union/src/main/kotlin/org/stepik/plugin/projectWizard/ui/ProjectSetting.kt
package org.stepik.plugin.projectWizard.ui

import org.stepik.api.objects.StudyObject
import org.stepik.core.SupportedLanguages


internal interface ProjectSetting {
    fun selectedStudyNode(studyObject: StudyObject)

    fun addListener(listener: ProjectSettingListener)

    fun removeListener(listener: ProjectSettingListener)

    fun selectedProgrammingLanguage(language: SupportedLanguages)
}
